File: source/samples/order-pizza/util/pizza-responses.js
/** ********************************************************************************************************************
 *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                           *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICNSE-2.0                                                                     *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 ******************************************************************************************************************** */
/**
 * @author Solution Builders
 */

const formatPizzaMenu = require('./menu-formatter');
const {
  validPizzaSizes,
  validPizzaCountResponse,
  validPizzaCrust,
  validConfirmationResponse,
} = require('./valid-inputs');

console.debug('Loading pizza-responses module....');

/**
 * Contains valid responses, for the supported languages by the Bot, based on the order flow
 * @type {object}
 */
const pizzaBotResponses = {
  /**
   * Generates resounse for the supported languages if the customer has order history
   * @param {string} lang - Langugae of the Bot
   * @param {array<object>} menuItems - array of objects representing pizza menu items
   * @param {object} customerLastOrder - an object of the customer's last order
   * @param {boolean} hasOrdersAnsNo -  flag used to indecate the customer has order history but answred no to the quesion: would you like to order the same order as the last one?
   * @returns {string} response - response to be displayed
   */
  STEP_0: (lang, menuItems, customerLastOrder, hasOrdersAnsNo = false) => {
    let response;
    if (customerLastOrder !== undefined) {
      const languageMessage = {
        'en-US': `Welcome back to our Pizza Ordering Service. Would you like to order the same order as your last one?. Type: ${customerLastOrder.pizzaType}, Size: ${customerLastOrder.pizzaSize}, Number of Pizzas: ${customerLastOrder.pizzaCount}, and Crust: ${customerLastOrder.pizzaCrust}, (yes or no)?`,
        'pt-BR': `Bem-vindo de volta ao nosso serviço de pedidos de pizza. Você gostaria de pedir o mesmo pedido do último?. Tipo: ${customerLastOrder.pizzaType}, Tamanho: ${customerLastOrder.pizzaSize}, Quantidade de Pizzas: ${customerLastOrder.pizzaCount}, e Borda: ${customerLastOrder.pizzaCrust}, (sim ou não)?`,
        'es-US': `Bienvenido de nuevo a nuestro servicio de pedidos de pizza. ¿Le gustaría hacer el mismo pedido que el último?. Tipo: ${customerLastOrder.pizzaType}, Size: ${customerLastOrder.pizzaSize}, Number of Pizzas: ${customerLastOrder.pizzaCount}, and Crust: ${customerLastOrder.pizzaCrust}, (si o no)?`,
        'fr-FR': `Bienvenue à notre service de commande de pizza. Souhaitez-vous commander la même commande que votre dernière?. Type: ${customerLastOrder.pizzaType}, Size: ${customerLastOrder.pizzaSize}, Number of Pizzas: ${customerLastOrder.pizzaCount}, and Crust: ${customerLastOrder.pizzaCrust}, (Oui ou non)?`,
        'it-IT': `Bentornati al nostro servizio di ordinazione di pizze. Vorresti ordinare lo stesso ordine del tuo ultimo?. Tipo: ${customerLastOrder.pizzaType}, Size: ${customerLastOrder.pizzaSize}, Number of Pizzas: ${customerLastOrder.pizzaCount}, and Crust: ${customerLastOrder.pizzaCrust}, (sì o no)?`,
        'de-DE': `Willkommen zurück bei unserem Pizza-Bestellservice. Möchten Sie die gleiche Bestellung wie Ihre letzte bestellen?. Art: ${customerLastOrder.pizzaType}, Size: ${customerLastOrder.pizzaSize}, Number of Pizzas: ${customerLastOrder.pizzaCount}, and Crust: ${customerLastOrder.pizzaCrust}, (ja oder Nein)?`,
        'ru-RU': `Добро пожаловать в нашу Службу заказа пиццы. Вы хотите заказать тот же заказ, что и ваш последний?. Тип: ${customerLastOrder.pizzaType}, Размер: ${customerLastOrder.pizzaSize}, Количество пицц: ${customerLastOrder.pizzaCount}, а также Корочка: ${customerLastOrder.pizzaCrust}, (да или нет)?`,
      };

      response = languageMessage[lang];
      return response;
    }
    return formatPizzaMenu(lang, menuItems, hasOrdersAnsNo);
  },

  /**
   * Contains STEP 2 responses of the order flow for the supported languages by the Bot
   * @type {object}
   */
  STEP_2: {
    'en-US':
      'What size would you like, (small, medium, large, or extra-large)?',
    'pt-BR':
      'Qual tamanho você gostaria, (pequeno, médio, grande ou extra-grande)?',
    'es-US':
      '¿Qué tamaño le gustaría (pequeño, mediano, grande o extra-grande)?',
    'fr-FR':
      'Quelle taille aimeriez-vous (petit, moyen, grand, ou très-grand)?',
    'it-IT': 'Che taglia vorresti, (piccola, media, grande o extra-grande)?',
    'de-DE': 'Welche Größe möchten Sie (klein, mittel, groß oder extra-groß)?',
    'ru-RU':
      'Какой размер вам нужен (маленький, средний, большой или очень-большой)?',
  },

  /**
   * Contains STEP 3 responses of the order flow for the supported languages by the Bot
   * @type {object}
   */
  STEP_3: {
    'en-US': 'How many pizzas would you like?',
    'pt-BR': 'Quantas pizzas você gostaria?',
    'es-US': '¿Cuántas pizzas te gustaría?',
    'fr-FR': 'Combien de pizzas souhaitez-vous?',
    'it-IT': 'Quante pizze vorresti?',
    'de-DE': 'Wie viele Pizzen möchten Sie?',
    'ru-RU': 'Сколько пиццы вы хотите?',
  },

  /**
   * Contains STEP 4 responses of the order flow for the supported languages by the Bot
   * @type {object}
   */
  STEP_4: {
    'en-US': 'What crust would you like, (thin or thick)?',
    'pt-BR': 'Que crosta você gostaria, (fina ou grossa)?',
    'es-US': '¿Qué corteza te gustaría, (fina o gruesa)?',
    'fr-FR': 'Quelle croûte aimeriez-vous (mince ou épaisse)?',
    'it-IT': 'Quale crosta vorresti, (sottile o spessa)?',
    'de-DE': 'Welche Kruste möchten Sie (dünn oder dick)?',
    'ru-RU': 'Какую корочку вы хотите (тонкий или толстую)?',
  },

  /**
   * Generates STEP 5 (order's summary) response for the supported languages by the Bot
   * @param {string} lang - Language of the Bot
   * @param {string} pizzaType - type of pizza requested by customer
   * @param {string} pizzaSize - size of pizza requested by customer
   * @param {string} pizzaCount - number of pizzas requested by customer
   * @param {string} pizzaCrust - crust of pizza requested by customer
   * @returns {string} - order's summary response
   */
  STEP_5: (lang, { pizzaType, pizzaSize, pizzaCount, pizzaCrust }) => {
    const languageMessage = {
      'en-US': `Here is a summary of your order. Type: ${pizzaType}, Size: ${pizzaSize}, Number of Pizzas: ${pizzaCount}, and Crust: ${pizzaCrust}. Would you like to place your order, (yes or no)?`,
      'pt-BR': `Aqui está um resumo do seu pedido. Tipo: ${pizzaType}, Tamanho: ${pizzaSize}, Quantidade de Pizzas: ${pizzaCount}, e Borda: ${pizzaCrust}. Você gostaria de fazer seu pedido, (sim ou não)?`,
      'es-US': `A continuación se muestra un resumen de su pedido. Tipo: ${pizzaType}, Tamaño: ${pizzaSize}, Numero de pizzas: ${pizzaCount}, y Corteza: ${pizzaCrust}. ¿Le gustaría realizar su pedido (si o no)?`,
      'fr-FR': `Voici un récapitulatif de votre commande. Type: ${pizzaType}, Taille: ${pizzaSize}, Nombre de pizzas: ${pizzaCount}, et croûte: ${pizzaCrust}. Souhaitez-vous passer votre commande (oui ou non)?`,
      'it-IT': `Ecco un riepilogo del tuo ordine. Tipo: ${pizzaType}, Taglia: ${pizzaSize}, Numero di pizze: ${pizzaCount}, e Crosta: ${pizzaCrust}. Vorresti effettuare l'ordine, (sì o no)?`,
      'de-DE': `Hier ist eine Zusammenfassung Ihrer Bestellung. Art: ${pizzaType}, Größe: ${pizzaSize}, Anzahl der Pizzen: ${pizzaCount}, und Kruste: ${pizzaCrust}. Möchten Sie Ihre Bestellung aufgeben (ja oder nein)?`,
      'ru-RU': `Вот краткое изложение вашего заказа. Тип: ${pizzaType}, Размер: ${pizzaSize}, Количество пицц: ${pizzaCount}, а также Корочка: ${pizzaCrust}. Вы бы хотели разместить заказ (да или нет)?`,
    };

    return languageMessage[lang];
  },

  /**
   * Generates STEP 6 (order confirmation) response for the supported languages by the Bot
   * @param {string} lang - Language of the Bot
   * @param {string} orderId - order's id generated by the Bot
   * @param {number} totalBill - order's total bill
   * @param {string} confirmation - order's confirmation response provided by customer to place the order (e.g. yes/no)
   * @returns {string} - order's confirmation response
   */
  STEP_6: ({ lang, orderId, totalBill, confirmation }) => {
    const languageYesRespones = {
      'en-US': `Your order has been placed. Here is the order's number: ${orderId}. Your total bill, including tax, is $${totalBill}. Thank you for using our service!`,
      'pt-BR': `Seu pedido foi feito. Aqui está o número do pedido: ${orderId}. Sua fatura total incluindo impostos é $${totalBill}. Obrigado por usar o nosso serviço!`,
      'es-US': `Su orden ha sido puesta. Aquí está el número de pedido: ${orderId}. Su factura total, incluidos los impuestos, es $${totalBill}. ¡Gracias por usar nuestro servicio!`,
      'fr-FR': `Votre commande a bien été reçue. Voici le numéro de commande: ${orderId}. Votre facture totale, taxes comprises, est $${totalBill}. Merci d'utiliser notre service!`,
      'it-IT': `Il tuo ordine è stato inoltrato. Ecco il numero dell'ordine: ${orderId}. Il conto totale comprensivo di tasse è $${totalBill}. Grazie per aver utilizzato il nostro servizio!`,
      'de-DE': `Deine Bestellung wurde aufgenommen. Hier ist die Bestellnummer: ${orderId}. Ihre Gesamtrechnung einschließlich Steuern beträgt $${totalBill}. Vielen Dank, dass Sie unseren Service nutzen!`,
      'ru-RU': `Ваш заказ был размещен. Вот номер заказа: ${orderId}. Ваш общий счет, включая налоги, составляет $${totalBill}. Спасибо, что воспользовались нашим сервисом!`,
    };

    const languageNoRespones = {
      'en-US': 'Your order has been cancelled. Thank you!',
      'pt-BR': 'Seu pedido foi cancelado. Obrigado!',
      'es-US': 'Tu pedido ha sido cancelado. ¡Gracias!',
      'fr-FR': 'Votre commande a été annulée. Je vous remercie!',
      'it-IT': 'Il tuo ordine è stato annullato. Grazie!',
      'de-DE': 'Ihre Bestellung wurde storniert. Dankeschön!',
      'ru-RU': 'Ваш заказ был отменен. Спасибо!',
    };

    const confirmationMessage = confirmation.trim().toLowerCase();
    /** If confirmation is yes */
    if (confirmationMessage === validConfirmationResponse[lang][0]) {
      return languageYesRespones[lang];
    }
    /** If confirmation is no */
    if (confirmationMessage === validConfirmationResponse[lang][1]) {
      return languageNoRespones[lang];
    }
    throw new Error(
      `pizza-responses.STEP_6: Unknown confirmationMessage: ${confirmationMessage}`
    );
  },
  /**
   * Contains Cancel responses of the order flow for the supported languages by the Bot, where the customer requested to cancel and end the order flow
   * @type {object}
   */
  CANCEL: {
    'en-US': 'The order flow has been canceled as requested. Thank you!',
    'pt-BR': 'O fluxo do pedido foi cancelado conforme solicitado. Obrigado!',
    'es-US':
      'El flujo de pedidos se ha cancelado según lo solicitado. ¡Gracias!',
    'fr-FR':
      'Le flux de commande a été annulé comme demandé. Je vous remercie!',
    'it-IT': 'Il flusso dell ordine è stato annullato come richiesto. Grazie!',
    'de-DE': 'Der Auftragsfluss wurde wie gewünscht storniert. Dankeschön!',
    'ru-RU': 'Поток заказов был отменен по запросу. Спасибо!',
  },
};

/**
 *
 * @param {string} id - id of the order's floe step, used by the Bot to collect the response from the customer
 * @param {string} text - message to be displayed by the Bot to the customer to request order's details or ask for valid inputs
 * @param {string} speech - message to be spoked by Polly to request order's details or ask for valid inputs (mostly same as text)
 * @param {object} payload - the modified event based by the Core lambda function
 * @param {boolean} endConversation - a flag used to inform the Core lambda that the pizza order flow is complete (true ends the flow)
 * @returns {object} - object based back to the Core lambda, containing order's context and other information as expected by the Core lambda
 */
const createResponse = (
  id,
  text,
  speech,
  payload,
  endConversation = false
) => ({
  asyncConversation: {
    id,
    payload,
    endConversation,
    ask: { text, speech },
  },
});

module.exports = {
  pizzaBotResponses,
  createResponse,
  validPizzaSizes,
  validPizzaCrust,
  validConfirmationResponse,
  validPizzaCountResponse,
};

File: source/samples/lex-lambdas/pizza_order/pizza_responses.py
######################################################################################################################
#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                #
#                                                                                                                    #
#  Licensed under the Apache License, Version 2.0 (the 'License'). You may not use this file except in compliance    #
#  with the License. A copy of the License is located at                                                             #
#                                                                                                                    #
#      http://www.apache.org/licenses/LICENSE-2.0                                                                    #
#                                                                                                                    #
#  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES #
#  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    #
#  and limitations under the License.                                                                                #
######################################################################################################################

# @author Solution Builders
import json

def get_menu_message(locale_id, welcome_message=True):
    '''
    Constructs a response containing a welcome message and pizza order menu for Amazon Lex to present to the user.

    :param locale_id: A string containig language locale id received from Amazon Lex with format example: en_US
    :param welcome_message: (optional) A boolean indicating whether the message should include a welcome message.
    :returns: A string in the language indicated by locale_id, containing order pizza menu and potentially a welcome message.
    '''

    welcome_responses = {
        'en_US': 'Welcome to our Pizza Ordering Service. ',
        'fr_FR': 'Bienvenue dans notre service de commande de pizza. ',
        'es_US': 'Bienvenido a nuestro servicio de pedidos de pizza. ',
        'it_IT': 'Benvenuti nel nostro servizio di ordinazione di pizze. ',
        'de_DE': 'Willkommen bei unserem Pizza-Bestellservice. ',
        'ja_JP': 'ピザ注文サービスへようこそ。'
    }
    menu_responses = {
        'en_US': 'Our Pizza menu includes: ',
        'fr_FR': 'Notre menu Pizza comprend: ',
        'es_US': 'Nuestro menú de Pizza incluye: ',
        'it_IT': 'Il nostro menù Pizza comprende: ',
        'de_DE': 'Unsere Pizza-Speisekarte beinhaltet: ',
        'ja_JP': 'ピザメニューには以下が含まれます: '
    }
    question_responses = {
        'en_US': 'What type of pizza would you like?',
        'fr_FR': 'Quel type de pizza souhaitez-vous?',
        'es_US': 'Que tipo de pizza te gustaria?',
        'it_IT': 'Che tipo di pizze vorresti?',
        'de_DE': 'Welche Art von Pizze möchten Sie?',
        'ja_JP': 'どのピザを注文されますか？'
    }
    with open("pizza_order/pizza_menu.json") as menu_file:
        menu_data = json.load(menu_file)
        response = ""
        if welcome_message:
            response = welcome_responses[locale_id]
        response = response + menu_responses[locale_id]
        for item in menu_data[locale_id]["menuItems"]:
            if locale_id == "en_US":
                response = response + f"{item['T']} Pizza ({item['D']}). Price (Small: {item['P']['small']}, Medium: {item['P']['medium']}, Large: {item['P']['large']}, Extra-large: {item['P']['extra-large']}). "
            elif locale_id == "fr_FR":
                response = response + f"{item['T']} Pizza ({item['D']}). Prix (Petit: {item['P']['petit']}, Moyen: {item['P']['moyen']}, Grand: {item['P']['grand']}, Très grand: {item['P']['très-grand']}). "
            elif locale_id == "es_US":
                response = response + f"{item['T']} Pizza ({item['D']}). Precio (Pequeña: {item['P']['pequeño']}, Mediana: {item['P']['mediano']}, Grande: {item['P']['grande']}, Extra-grande: {item['P']['extra-grande']}). "
            elif locale_id == "it_IT":
                response = response + f"{item['T']} Pizza ({item['D']}). Prezzo (Piccola: {item['P']['piccola']}, Media: {item['P']['media']}, Grande: {item['P']['grande']}, Extra-grande: {item['P']['extra-grande']}). "
            elif locale_id == "de_DE":
                response = response + f"{item['T']} Pizza ({item['D']}). Preis (Kleine: {item['P']['klein']}, Mittlere: {item['P']['mittel']}, Große: {item['P']['groß']}, Extra-große: {item['P']['extra-groß']}). "
            elif locale_id == "ja_JP":
                response = response + f"{item['T']} ピザ ({item['D']})。 価格 (S: {item['P']['S']}, M: {item['P']['M']}, L: {item['P']['L']}, XL: {item['P']['XL']})。 "
            else:
                raise ValueError(f"Unsupported locale id. Parameter `locale_id: {locale_id}` is not supported.")

        response = response + question_responses[locale_id]
        return response


def get_repeat_message(locale_id, last_order):
    '''
    Constructs a response containing a welcome message and prmopt message to repeat the user's last order for Amazon Lex to present to the user.

    :param locale_id: A string containig language locale id received from Amazon Lex with format example: en_US
    :param last_order: 'Items' object received from DynamoDB query
    :returns: A string in the language indicated by locale_id, indicating whether user wants to repeat their last order.
    '''
    pizza_type = last_order['pizzaType']['S']
    pizza_size = last_order['pizzaSize']['S']
    pizza_crust = last_order['pizzaCrust']['S']
    pizza_count = last_order['pizzaCount']['N']
    repeat_messages = {
        'en_US': f'Welcome back to our Pizza Ordering Service. Would you like to order the same order as your last one?. Type: {pizza_type}, Size: {pizza_size}, Number of Pizzas: {pizza_count}, and Crust: {pizza_crust}, (yes or no)?',
        'es_US': f'Bienvenido de nuevo a nuestro servicio de pedidos de pizza. ¿Le gustaría hacer el mismo pedido que el último?. Tipo: {pizza_type}, Size: {pizza_size}, Number of Pizzas: {pizza_count}, and Crust: {pizza_crust}, (si o no)?',
        'fr_FR': f'Bienvenue à notre service de commande de pizza. Souhaitez-vous commander la même commande que votre dernière?. Type: {pizza_type}, Size: {pizza_size}, Number of Pizzas: {pizza_count}, and Crust: {pizza_crust}, (Oui ou non)?',
        'it_IT': f'Bentornati al nostro servizio di ordinazione di pizze. Vorresti ordinare lo stesso ordine del tuo ultimo?. Tipo: {pizza_type}, Size: {pizza_size}, Number of Pizzas: {pizza_count}, and Crust: {pizza_crust}, (sì o no)?',
        'de_DE': f'Willkommen zurück bei unserem Pizza-Bestellservice. Möchten Sie die gleiche Bestellung wie Ihre letzte bestellen?. Art: {pizza_type}, Size: {pizza_size}, Number of Pizzas: {pizza_count}, and Crust: {pizza_crust}, (ja oder Nein)?',
        'ja_JP': f'ピザ注文サービスへようこそ。 前回と同じ注文をしますか？ タイプ：{pizza_type}、サイズ：{pizza_size}、ピザの数：{pizza_count}、クラスト：{pizza_crust}、（はいまたはいいえ）？',
    }
    return repeat_messages[locale_id]


def get_confirmation_message(locale_id, slots):
    '''
    Constructs a response containing order summary and prompt for confirmation for Amazon Lex to present to the user.

    :param locale_id: A string containig language locale id received from Amazon Lex with format example: en_US
    :param last_order: "Items" object received from DynamoDB query
    :returns: A string in the language indicated by locale_id, indicating whether user wants to repeat their last order.
    '''
    pizza_type = slots['type']['value']['resolvedValues'][0]
    pizza_size = slots['size']['value']['resolvedValues'][0]
    pizza_count = slots['count']['value']['resolvedValues'][0]
    pizza_crust = slots['crust']['value']['resolvedValues'][0]
    confirmation_responses = {
        'en_US': f'Here is a summary of your order. Type: {pizza_type}, Size: {pizza_size}, Number of Pizzas: {pizza_count}, and Crust: {pizza_crust}. Would you like to place your order, (yes or no)?',
        'es_US': f'A continuación se muestra un resumen de su pedido. Tipo: {pizza_type}, Tamaño: {pizza_size}, Numero de pizzas: {pizza_count}, y Corteza: {pizza_crust}. ¿Le gustaría realizar su pedido (si o no)?',
        'fr_FR': f'Voici un récapitulatif de votre commande. Type: {pizza_type}, Taille: {pizza_size}, Nombre de pizzas: {pizza_count}, et croûte: {pizza_crust}. Souhaitez-vous passer votre commande (oui ou non)?',
        'it_IT': f"Ecco un riepilogo del tuo ordine. Tipo: {pizza_type}, Taglia: {pizza_size}, Numero di pizze: {pizza_count}, e Crosta: {pizza_crust}. Vorresti effettuare l'ordine, (sì o no)?",
        'de_DE': f'Hier ist eine Zusammenfassung Ihrer Bestellung. Art: {pizza_type}, Größe: {pizza_size}, Anzahl der Pizzen: {pizza_count}, und Kruste: {pizza_crust}. Möchten Sie Ihre Bestellung aufgeben (ja oder nein)?',
        'ja_JP': f'こちらがご注文の概要です。 タイプ：{pizza_type}、サイズ：{pizza_size}、ピザの数：{pizza_count}、クラスト：{pizza_crust}。 注文しますか (はい、いいえ)？ ',
    }
    return confirmation_responses[locale_id]


def get_fulfilled_message(locale_id, order_id, total_bill):
    '''
    Constructs a response containing an order fulfillment message for Amazon Lex to present to the user.

    :param locale_id: A string containig language locale id received from Amazon Lex with format example: en_US
    :param order_id: A string containing a generated order id
    :param total_bill: A string containing number value of the total bill
    :returns: A string in the language indicated by locale_id, informing the user that their order has been placed.
    '''
    fufilled_messages = {
      'en_US': f"Your order has been placed. Here is the order's number: {order_id}. Your total bill, including tax, is ${total_bill}. Thank you for using our service!",
      'es_US': f'Su orden ha sido puesta. Aquí está el número de pedido: {order_id}. Su factura total, incluidos los impuestos, es ${total_bill}. ¡Gracias por usar nuestro servicio!',
      'fr_FR': f"Votre commande a bien été reçue. Voici le numéro de commande: {order_id}. Votre facture totale, taxes comprises, est ${total_bill}. Merci d'utiliser notre service!",
      'it_IT': f"Il tuo ordine è stato inoltrato. Ecco il numero dell'ordine: {order_id}. Il conto totale comprensivo di tasse è ${total_bill}. Grazie per aver utilizzato il nostro servizio!",
      'de_DE': f'Deine Bestellung wurde aufgenommen. Hier ist die Bestellnummer: {order_id}. Ihre Gesamtrechnung einschließlich Steuern beträgt ${total_bill}. Vielen Dank, dass Sie unseren Service nutzen!',
      'ja_JP': f"ご注文は完了しました。 注文番号は次のとおりです：{order_id}。 税込みの合計請求額は$ {total_bill}です。 私たちのサービスをご利用いただきありがとうございます！",
    }
    return fufilled_messages[locale_id]


def get_cancel_message(locale_id):
    '''
    Constructs a response containing a cancellation message for Amazon Lex to present to the user.

    :param locale_id: A string containig language locale id received from Amazon Lex with format example: en_US
    :returns: A string in the language indicated by locale_id, informing the user that their order has been cancelled.
    '''
    cancel_responses = {
        "en_US": "Your order has been cancelled. Thank you!",
        "es_US": "Tu pedido ha sido cancelado. ¡Gracias!",
        "fr_FR": "Votre commande a été annulée. Je vous remercie!",
        "it_IT": "Il tuo ordine è stato annullato. Grazie!",
        "de_DE": "Ihre Bestellung wurde storniert. Dankeschön!",
        "ja_JP": "ご注文はキャンセルされました。 ありがとうございました！ ",
    }
    return cancel_responses[locale_id]